src/library/HRGEventGenerator/HypersurfaceSampler.cpp

```diff
@@ -635,16 +641,24 @@ namespace thermalfist {
         }
         // this is in principle the ansatz which is also used in https://github.com/smash-transport/smash-hadron-sampler
         // from this paper Phys.Rev.C 73 (2006) 064903
-        Weight_visc = (1.0 + (1.0 + stat * feq) * pipp / (2.0 * T * T * (elem->edens + elem->P)));
+        Weight_visc += ((1.0 + stat * feq) * pipp / (2.0 * T * T * (elem->edens + elem->P)));
+        
+      }
+      if (bulk_correction){
+        // this is in principle the ansatz which is also used in https://github.com/smash-transport/smash-hadron-sampler
+        // eq (7) of https://arxiv.org/pdf/1509.06738
+         double mom[4] = {part.p0, part.px, part.py, part.pz};
+         Weight_visc -= (1.0+stat*feq)*elem->Pi*(mass*mass/(3*mom[0])-mom[0]*(1.0/3.0-speed_of_sound_squared))//what to do about this
```

Comment:
This is the expression from Iurii's [note](https://github.com/smash-transport/smash-hadron-sampler/files/14011233/bulk_corrections_note.pdf)